### PR TITLE
feat: Add per table sync metrics file flag

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Goes with https://github.com/cloudquery/plugin-sdk/pull/1751

Opening for reference

### CODE REVIEW

File: cli/cmd/sync.go [hunk 1]
```diff
@@ -123,7 +124,16 @@ func sync(cmd *cobra.Command, args []string) error {
 	// in a cloud sync environment, we pass only the relevant environment variables to the plugin
 	osEnviron := os.Environ()
 
+	otelReceiver, err := otel.StartOtelReceiver(ctx)
+	if err == nil {
+		defer otelReceiver.Shutdown(ctx)
+	}
+
 	for _, source := range sources {
+		if source.OtelEndpoint == "" {
```
Comment (erezrokah) [inline cli/cmd/sync.go:133]: Probably need to be a spec option instead of the default

File: cli/internal/otel/receiver.go [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+	if c.tracesFile != nil {
+		err = errors.Join(err, c.tracesFile.Close())
+	}
+	if c.metricsFile != nil {
+		err = errors.Join(err, c.metricsFile.Close())
+	}
+	return err
+}
+
+func newDefaultSpanConsumer(tracesFile *os.File) ConsumeSpan {
```
Comment (erezrokah) [inline cli/internal/otel/receiver.go:65]: I'm writing spans and metrics to files but we could do pretty much what we want we the data

File: cli/internal/otel/receiver.go [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+	consumeMetric   ConsumeMetric
+}
+
+func (c *Consumer) Shutdown(ctx context.Context) error {
+	var err error
+	if c.quit != nil {
+		close(c.quit)
+	}
+	if c.metricsFile != nil {
+		err = errors.Join(err, c.metricsFile.Close())
```
Comment (marianogappa) [inline cli/internal/otel/receiver.go:73]: This joins ` c.metricsFile.Close()` with `nil`. I assume it was a rewrite leftover?

File: cli/internal/otel/receiver.go [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+func newDefaultSpanConsumer() ConsumeSpan {
+	return func(ctx context.Context, span Span) {
+		// do nothing
+	}
+}
+
+func newDefaultMetricConsumer(metricsFile *os.File, quit chan any) ConsumeMetric {
+	tableLock := sync.Mutex{}
+	metricsMap := make(map[string]*TableMetric)
+	ticker := time.NewTicker(20 * time.Second)
```
Comment (marianogappa) [inline cli/internal/otel/receiver.go:87]: Should this be configurable?

File: cli/internal/otel/receiver.go [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+				metrics.Table,
+				metrics.ClientId,
+				metrics.StartTime,
+				metrics.EndTime,
+				metrics.Resources,
+				metrics.Errors,
+				metrics.Panics,
+			}
+			if metrics.EndTime == nil {
+				row[3] = "N/A"
```
Comment (marianogappa) [inline cli/internal/otel/receiver.go:125]: I'm aware that `row`'s type is `[]interface{}`, but unsure how the code that uses it works. Is it ok to mix `EndTime *time.Time` with `string` on `row[3]`?

**Review by marianogappa — APPROVED** [review]
I added a few comments. It's not a bulletproof review because I'm a little unfamiliar with Go's otel (I've used it in Python). I assume you've tested this code and it largely works.

File: cli/internal/otel/receiver.go (hunk 5 above)
Comment (erezrokah) [reply cli/internal/otel/receiver.go:125]: Or so without this the table with render `<nil>` for nil values, I'll post an example for how the content of the file looks like in a bit

File: cli/internal/otel/receiver.go (hunk 3 above)
Comment (erezrokah) [reply cli/internal/otel/receiver.go]: Fixed in https://github.com/cloudquery/cloudquery/pull/18397/commits/0f79d15bac686b591c2de5e0192d9a42075fe349

File: cli/internal/otel/receiver.go (hunk 4 above)
Comment (erezrokah) [reply cli/internal/otel/receiver.go:85]: Could be, not sure how to expose it yet? Another CLI flag seems like an overkill

**Comment by erezrokah** [comment]
This works only with plugins released with https://github.com/cloudquery/plugin-sdk/pull/1751 so I'll wait before merging this

**Comment by erezrokah** [comment]
> This works only with plugins released with [cloudquery/plugin-sdk#1751](https://github.com/cloudquery/plugin-sdk/pull/1751) so I'll wait before merging this

OK so I'll actually mark this as a hidden flag so I can release the CLI, then unhide it once all plugins are updated

**Pull request merged**